spec/integration/apply_full_spec.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'bolt_spec/conn'
+require 'bolt_spec/files'
+require 'bolt_spec/integration'
+
+describe "apply" do
+  include BoltSpec::Conn
+  include BoltSpec::Files
+  include BoltSpec::Integration
+
+  let(:modulepath) { File.join(__dir__, '../fixtures/apply') }
+  let(:config_flags) { %W[--format json --nodes #{uri} --password #{password} --modulepath #{modulepath}] + tflags }
+
+  describe 'over ssh', ssh: true do
+    let(:uri) { conn_uri('ssh') }
+    let(:password) { conn_info('ssh')[:password] }
+    let(:apply_task) { 'apply_catalog.sh' }
+    let(:tflags) { %W[--no-host-key-check --run-as root --sudo-password #{password}] }
+
+    after(:each) do
+      uninstall = '/opt/puppetlabs/bin/puppet resource package puppet-agent ensure=absent'
+      run_cli_json(%W[command run #{uninstall}] + config_flags)
+    end
+
+    it 'installs puppet' do
```

Comment:
Right now this is pretty simple, but it's something we can probably build on. I think for this PR its sufficient.